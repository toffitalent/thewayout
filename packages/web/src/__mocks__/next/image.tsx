import * as React from 'react';

// eslint-disable-next-line react/jsx-no-useless-fragment
const mock = (props: { children: React.ReactElement }): React.ReactElement => <>{props.children}</>;

export default mock;
