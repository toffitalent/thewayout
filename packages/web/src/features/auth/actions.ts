import { CreateUserRequest } from '@two/shared';
import { API } from '@app/api';
import { createApiThunk } from '@app/utils';

export const fetchUser = createApiThunk('auth/fetchUser', API.user.retrieve);
export const forgotPassword = createApiThunk('auth/forgotPassword', API.user.forgotPassword);

export const login = createApiThunk(
  'auth/login',
  async (credentials: Parameters<typeof API.login>[0], { dispatch }) => {
    await API.login(credentials);
    await dispatch(fetchUser());
  },
);

export const logout = createApiThunk('auth/logout', API.logout);

export const resetPassword = createApiThunk(
  'auth/resetPassword',
  async (
    { username, password, token }: { username: string; password: string; token: string },
    { dispatch },
  ) => {
    await API.user.resetPassword(password, token);
    return dispatch(login({ username, password }));
  },
);

export const signUp = createApiThunk(
  'auth/register',
  async (payload: CreateUserRequest, { dispatch }) => {
    await API.user.create(payload);
    return dispatch(login({ username: payload.email, password: payload.password }));
  },
);

export const updateUser = createApiThunk('auth/updateUser', API.user.update);
