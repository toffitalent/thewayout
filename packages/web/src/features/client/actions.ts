import { ListRspClientRequest } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createClient = createApiThunk('client/create', API.clientProfile.create);

export const updateClient = createApiThunk('client/update', API.clientProfile.update);

export const listAllJobs = createApiThunk(
  'client/jobs',
  ({ forceReload }: { forceReload?: boolean }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.jobsList;
    const lastId = [...ids].pop() as string;
    return API.job.list({
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const retrieveJob = createApiThunk(
  'client/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const applyJob = createApiThunk(
  'client/jobApply',
  ({ jobId, questions }: { jobId: string; questions?: { [key: string]: string }[] }) =>
    API.job.apply(jobId, {
      ...(questions ? { questions } : {}),
    }),
);

export const listAppliedJobs = createApiThunk(
  'client/applications',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).clientJobs.appliedJobs;
    const lastId = [...ids].pop() as string;
    return API.clientProfile.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const listSuggestedJobs = createApiThunk('client/suggestedJobs', (clientId: string) =>
  API.clientProfile.listSuggestedJobs(clientId),
);

export const listRsp = createApiThunk(
  'rsp/list',
  ({ userId, ...rest }: ListRspClientRequest & { userId: string }) =>
    API.clientProfile.listRsp({ userId, ...rest }),
);

export const retrieveCaseManager = createApiThunk(
  'client/caseManager',
  API.clientProfile.getCaseManager,
);
