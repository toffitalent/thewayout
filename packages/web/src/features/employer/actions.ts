import { Job, JobApplicationStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createEmployer = createApiThunk('employer/create', API.employer.create);

export const updateEmployer = createApiThunk('employer/update', API.employer.update);

export const createJob = createApiThunk('employer/job/create', API.job.create);

export const listJobs = createApiThunk(
  'employer/jobs',
  ({ forceReload, userId }: { forceReload?: boolean; userId: string }, { getState }) => {
    const { ids } = (getState() as State).employerJobs.jobs;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      ...(!forceReload && { before: lastId }),
    });
  },
);

export const updateListJobs = createApiThunk(
  'employer/jobs/listUpdated',
  ({ userId }: { userId: string }, { getState }) => {
    const { listUpdatedAt, ids } = (getState() as State).employerJobs.jobs;
    const firstId = ids[0] as string;
    const lastId = [...ids].pop() as string;
    return API.employer.listJobs(userId, {
      since: listUpdatedAt,
      after: firstId,
      before: lastId,
    });
  },
);

export const retrieveJob = createApiThunk(
  'employer/job',
  ({ jobId, since }: { jobId: string; since?: Date }) =>
    API.job.retrieve(jobId, {
      ...(since ? { since } : {}),
    }),
);

export const listApplications = createApiThunk(
  'employer/listApplications',
  ({ employerId, jobId }: { employerId: string; jobId: string }, { getState }) => {
    const { entities } = (getState() as State).employerJobs.jobsApplications;
    const page = entities[jobId]?.page;
    const isListEmpty = !entities[jobId]?.applications.length;

    return API.employer.listApplications(employerId, jobId, {
      page: isListEmpty ? 0 : page! + 1,
    });
  },
);

export const getClient = createApiThunk(
  'employer/getClient',
  ({ employerId, jobId, clientId }: { employerId: string; jobId: string; clientId: string }) =>
    API.employer.getClient(employerId, jobId, clientId, {}),
);

export const updateJob = createApiThunk(
  'employer/updateJob',
  ({ jobId, patch }: { jobId: string; patch: Partial<Job> }) => API.job.update(jobId, patch),
);

export const interviewClient = createApiThunk(
  'employer/interviewClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.interview,
    ),
);

export const hireClient = createApiThunk(
  'employer/hireClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.hired,
    ),
);

export const rejectClient = createApiThunk(
  'employer/rejectClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.rejected,
    ),
);

export const notAFitClient = createApiThunk(
  'employer/notAFitClient',
  ({
    employerId,
    jobId,
    applicationId,
  }: {
    employerId: string;
    jobId: string;
    applicationId: string;
  }) =>
    API.employer.changeApplicationStatus(
      employerId,
      jobId,
      applicationId,
      JobApplicationStatus.notAFit,
    ),
);
