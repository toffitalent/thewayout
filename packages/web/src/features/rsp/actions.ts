import { ClientType, RspClientStatus } from '@two/shared';
import { API } from '@app/api';
import type { State } from '@app/store';
import { createApiThunk } from '@app/utils';

export const createRsp = createApiThunk('rsp/create', API.rsp.create);

export const updateRsp = createApiThunk('rsp/update', API.rsp.update);

export const createMember = createApiThunk('rsp/createMember', API.rsp.createMember);

export const inviteCaseManager = createApiThunk('rsp/inviteCaseManager', API.rsp.inviteCaseManager);

export const removeInvitation = createApiThunk('rsp/removeInvitation', API.rsp.removeInvitation);

export const removeCaseManager = createApiThunk('rsp/removeCaseManager', API.rsp.removeCaseManager);

export const listCaseManagers = createApiThunk(
  'rsp/caseManagers',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.caseManagers;
    const lastId = [...ids].pop() as string;
    return API.rsp.listCaseManagers(rspId, {
      before: lastId,
    });
  },
);

export const listInvitations = createApiThunk(
  'rsp/invitations',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.invitations;
    const lastId = [...ids].pop() as string;
    return API.rsp.listInvitations(rspId, {
      before: lastId,
    });
  },
);

export const listCaseManagersNames = createApiThunk(
  'rsp/listCaseManagersNames',
  API.rsp.listCaseManagersNames,
);

export const retrieveRspAccount = createApiThunk('rsp/retrieveRspAccount', API.rsp.retrieveAccount);

export const updateCaseManager = createApiThunk('rsp/updateCaseManager', API.rsp.updateCaseManager);

export const listClients = createApiThunk(
  'rsp/clients',
  ({ rspId }: { rspId: string }, { getState }) => {
    const { ids } = (getState() as State).rsp.clients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      before: lastId,
    });
  },
);

export const listStatisticsClients = createApiThunk(
  'rsp/statisticsClients',
  (
    {
      rspId,
      // eslint-disable-next-line @typescript-eslint/no-unused-vars
      forceReload,
      ...clientFilters
    }: Partial<ClientType> & { rspId: string; forceReload?: boolean },
    { getState },
  ) => {
    const { ids } = (getState() as State).rsp.statisticsClients;
    const lastId = [...ids].pop() as string;
    return API.rsp.listClients(rspId, {
      ...clientFilters,
      before: lastId,
    });
  },
);

export const retrieveClient = createApiThunk('rsp/retrieveClient', API.rsp.retrieveClient);

export const declineClient = createApiThunk(
  'rsp/declineClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.closed),
);

export const acceptClient = createApiThunk(
  'rsp/acceptClient',
  ({ rspId, clientId }: { rspId: string; clientId: string }) =>
    API.rsp.changeClientStatus(rspId, clientId, RspClientStatus.active),
);

export const addClientNotes = createApiThunk('rsp/addClientNotes', API.rsp.addClientNotes);

export const assignCaseManager = createApiThunk(
  'rsp/assignCaseManager',
  ({
    rspId,
    clientId,
    caseManagerId,
  }: {
    rspId: string;
    clientId: string;
    caseManagerId: string;
  }) => API.rsp.assignCaseManager(rspId, clientId, caseManagerId),
);

export const getStatistics = createApiThunk('rsp/statistics', API.rsp.getStatistics);
