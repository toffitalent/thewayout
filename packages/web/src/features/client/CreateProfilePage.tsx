import { Box, Toast, Wizard } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { getClientDataFromWizard } from '@app/utils';
import { selectAuthUser } from '../auth';
import { createClient } from './actions';
import styles from './CreateProfilePage.module.scss';
import { steps } from './profileData';
import { WizardData } from './profileDataTypes';

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator();
  const router = useRouter();

  const handleSave = (data: WizardData) => {
    dispatch(createClient(getClientDataFromWizard(data, true)))
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Profile added!');
      });
  };

  if (!user) {
    return null;
  }
  if (user.client) {
    router.push('/client/');
    return null;
  }

  const lastStep = steps.pop();
  if (lastStep) {
    lastStep.nextStep = (data) => handleSave(data as WizardData);
    steps.push(lastStep);
  }

  return (
    <>
      <SEO title="Create Profile" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard steps={steps} layout={WizardLayout} />
      </Box>
    </>
  );
};
