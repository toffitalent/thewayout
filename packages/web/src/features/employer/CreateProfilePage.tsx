import { Box, Wizard, WizardStep } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { useCallback } from 'react';
import { EmployerCreateRequest as WizardData } from '@two/shared';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { selectAuthUser } from '../auth';
import { createEmployer } from './actions';
import { ConfirmationStep } from './ConfirmationStep';
import styles from './CreatorProfile.module.scss';
import { steps } from './profileData';

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const router = useRouter();
  const dispatch = useAppDispatch();
  const showError = useErrorIndicator();

  const handleSubmit = useCallback(
    (data: WizardData) => {
      if (user) {
        dispatch(createEmployer(data))
          .unwrap()
          .then(() => {
            router.push('/employer/create-profile/confirm/');
          })
          .catch((e) => {
            showError(e);
          });
      }
    },
    [dispatch, router, showError, user],
  );

  if (!user) {
    return null;
  }

  const confirmationStep: WizardStep = {
    id: 'confirm',
    title: 'Confirm your data',
    component: <ConfirmationStep onSubmit={handleSubmit} />,
    showBackButton: true,
  };

  return (
    <>
      <SEO title="Create Profile" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard steps={[...steps, confirmationStep]} layout={WizardLayout} />
      </Box>
    </>
  );
};
