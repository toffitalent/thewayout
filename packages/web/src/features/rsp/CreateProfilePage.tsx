import { Box, Toast, Wizard } from '@disruptive-labs/ui';
import { useRouter } from 'next/router';
import { RspRole, VeteranOrJustice } from '@two/shared';
import { SEO } from '@app/components/SEO';
import { WizardLayout } from '@app/components/WizardLayout';
import { useAppDispatch, useAppSelector, useErrorIndicator } from '@app/hooks';
import { selectAuthUser } from '../auth';
import { createMember, createRsp } from './actions';
import styles from './CreateProfilePage.module.scss';
import {
  memberProfileSteps,
  ownerProfileSteps,
  RspSteps,
  rspSteps,
  WizardRspData,
} from './rspData';

const getRspRequestData = (data: WizardRspData) => {
  const {
    phone,
    position,
    avatar,
    offenseDrugs,
    offenseMotorVehicle,
    offensePropertyDamage,
    offenseSexual,
    offenseTheft,
    offenseViolent,
    offenseWhiteCollar,
    phoneRspContact,
    emailRspContact,
    avatarRsp,
    ...rsp
  } = data;

  delete rsp.avatarUrl;
  delete rsp.avatarRspUrl;

  return {
    ...rsp,
    avatar: avatarRsp,
    phone: phoneRspContact,
    email: emailRspContact,
    offenses: [
      ...(offenseDrugs || []),
      ...(offensePropertyDamage || []),
      ...(offenseSexual || []),
      ...(offenseTheft || []),
      ...(offenseViolent || []),
      ...(offenseWhiteCollar || []),
      ...(offenseMotorVehicle || []),
    ],
    owner: { phone, position, avatar },
  };
};

export const CreateProfilePage = () => {
  const user = useAppSelector(selectAuthUser);
  const router = useRouter();
  const showError = useErrorIndicator({ ResourceConflict: 'Organization name already exist' });
  const dispatch = useAppDispatch();

  const handleSaveRsp = (data: WizardRspData) => {
    dispatch(createRsp(getRspRequestData(data)))
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Organization added!');
      });
  };

  const handleSaveMember = (data: WizardRspData) => {
    dispatch(
      createMember({
        ...data,
        userId: user?.id as string,
        rspId: user?.rspAccount?.rspId as string,
      }),
    )
      .unwrap()
      .catch((e) => {
        showError(e);
      })
      .then(() => {
        Toast.success('Profile added!');
      });
  };

  if (!user) {
    return null;
  }
  if (user.rspAccount?.isProfileFilled) {
    router.push(user.rspAccount.role === RspRole.owner ? '/ssp/case-managers' : '/ssp/clients');
    return null;
  }

  const isOwner = user.rspAccount?.role === RspRole.owner;

  let steps = memberProfileSteps;
  if (isOwner) {
    steps = [...memberProfileSteps, ...ownerProfileSteps, ...rspSteps];
  }

  if (isOwner) {
    const indexOfContact = steps.findIndex((el) => el.id === RspSteps.contact);
    steps[indexOfContact].nextStep = (data) => {
      if ((data as WizardRspData).veteranOrJustice.includes(VeteranOrJustice.justiceImpacted)) {
        return RspSteps.justiceStatus;
      }
      handleSaveRsp(data);
      return RspSteps.contact;
    };
  }

  const lastStep = steps.pop();
  if (lastStep) {
    lastStep.nextStep = (data) => (isOwner ? handleSaveRsp(data) : handleSaveMember(data));
    steps.push(lastStep);
  }

  return (
    <>
      <SEO title="" description="" />
      <Box className={styles.wizardWrapper}>
        <Wizard
          steps={steps}
          layout={WizardLayout}
          initialData={user.phone ? ({ phone: user.phone } as any) : undefined}
        />
      </Box>
    </>
  );
};
