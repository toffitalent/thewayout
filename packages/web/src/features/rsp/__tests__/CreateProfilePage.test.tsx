import userEvent from '@testing-library/user-event';
import { useRouter } from 'next/router';
import {
  JusticeStatus,
  justiceText,
  OffenseCategory,
  offenseText,
  RspPosition,
  rspPositionText,
  RspRole,
  State,
  states,
  Support,
  supportText,
  VeteranOrJustice,
} from '@two/shared';
import { act, fireEvent, fixtures, mockState, mockThunks, render, waitFor } from '@test';
import * as actions from '../actions';
import { CreateProfilePage } from '../CreateProfilePage';

const { user } = fixtures;

describe('CreateProfilePage', () => {
  let push: jest.Mock;

  beforeEach(() => {
    push = jest.fn();

    (useRouter as jest.Mock).mockImplementation(() => ({
      beforePopState: jest.fn(),
      push,
    }));
  });

  test('renders correctly', async () => {
    const { container, findByText } = render(<CreateProfilePage />, {
      initialState: {
        ...mockState,
        auth: { user: { ...user, rspAccount: { isProfileFilled: false } } },
      },
    });

    await waitFor(async () => expect(await findByText('Phone number')).toBeInTheDocument());
    expect(container).toMatchSnapshot();
  });

  test('redirects owner if rsp account is filled by user', async () => {
    render(<CreateProfilePage />, {
      initialState: {
        ...mockState,
        auth: { user: { ...user, rspAccount: { isProfileFilled: true, role: RspRole.owner } } },
      },
    });

    expect(push).toBeCalledWith('/ssp/case-managers');
  });

  test('redirects owner if rsp account is filled by user', async () => {
    render(<CreateProfilePage />, {
      initialState: {
        ...mockState,
        auth: { user: { ...user, rspAccount: { isProfileFilled: true, role: RspRole.member } } },
      },
    });

    expect(push).toBeCalledWith('/ssp/clients');
  });

  test('creates rsp organization and updates owner account', async () => {
    const [createRsp] = mockThunks(actions, ['createRsp']);

    const { findByRole, findByText, getAllByLabelText, dispatch } = render(<CreateProfilePage />, {
      initialState: {
        ...mockState,
        auth: { user: { ...user, rspAccount: { isProfileFilled: false, role: RspRole.owner } } },
      },
    });

    // Phone Number step
    await userEvent.type(await findByRole('textbox', { name: 'Phone number' }), '3344441224', {
      delay: 1,
    });
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Avatar step
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Role step
    await userEvent.selectOptions(
      await findByRole('combobox'),
      rspPositionText[RspPosition.operationManager],
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Organization Name step
    await userEvent.type(await findByRole('textbox', { name: 'Name' }), 'Test Organization', {
      delay: 1,
    });
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Rsp avatar step
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Organization Description step
    await userEvent.type(await findByRole('textbox', { name: 'Description' }), 'Test Description', {
      delay: 1,
    });
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Business Address step
    await userEvent.type(await findByRole('textbox', { name: 'Your Address' }), '111 Street', {
      delay: 1,
    });
    await userEvent.type(await findByRole('textbox', { name: 'Town / City' }), 'Los Angeles', {
      delay: 1,
    });
    await userEvent.selectOptions(await findByRole('combobox'), states[State.AK]);
    await userEvent.type(await findByRole('textbox', { name: 'Zip Code' }), '90111', {
      delay: 1,
    });
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Service Area step
    await userEvent.type(getAllByLabelText('Search Area')[0], 'W');
    await act(async () => fireEvent.click(await findByRole('option', { name: 'Washington' })));
    expect(await findByText('Selected Service Areas')).toBeInTheDocument();
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Veteran or Justice Impacted step
    await act(async () =>
      fireEvent.click(await findByRole('checkbox', { name: 'Justice Impacted' })),
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Support step
    await act(async () =>
      fireEvent.click(await findByRole('checkbox', { name: supportText[Support.employment] })),
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Organization Contact Info step
    await userEvent.type(
      await findByRole('textbox', { name: 'Phone Number (Optional)' }),
      '3344441223',
      {
        delay: 1,
      },
    );
    await userEvent.type(
      await findByRole('textbox', { name: 'Email (Optional)' }),
      'test@test.com',
      {
        delay: 1,
      },
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Accepted Justice Status step
    await act(async () =>
      fireEvent.click(
        await findByRole('checkbox', { name: justiceText[JusticeStatus.extendedSupervision] }),
      ),
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Allowed Offenses step
    await act(async () =>
      fireEvent.click(await findByRole('checkbox', { name: offenseText[OffenseCategory.arson] })),
    );
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    expect(createRsp).toBeCalledWith({
      name: 'Test Organization',
      description: 'Test Description',
      address: '111 Street',
      city: 'Los Angeles',
      state: State.AK,
      postalCode: '90111',
      servicesArea: ['Washington'],
      support: [Support.employment],
      justiceStatus: [JusticeStatus.extendedSupervision],
      offenses: [OffenseCategory.arson],
      phone: '3344441223',
      email: 'test@test.com',
      owner: {
        phone: '3344441224',
        position: RspPosition.operationManager,
        avatar: undefined,
      },
      veteranOrJustice: [VeteranOrJustice.justiceImpacted],
    });
    expect(dispatch).toBeCalled();
  }, 20000);

  test('creates member account', async () => {
    const [createMember] = mockThunks(actions, ['createMember']);

    const { findByRole, dispatch } = render(<CreateProfilePage />, {
      initialState: {
        ...mockState,
        auth: {
          user: {
            ...user,
            rspAccount: {
              id: 'TEST_ID',
              rspId: 'TEST_RSP_ID',
              isProfileFilled: false,
              role: RspRole.member,
            },
          },
        },
      },
    });

    // Phone Number step
    await userEvent.type(await findByRole('textbox', { name: 'Phone number' }), '3344441224', {
      delay: 1,
    });
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    // Avatar step
    await act(async () => fireEvent.click(await findByRole('button', { name: 'Save & Next' })));

    expect(createMember).toBeCalledWith({
      phone: '3344441224',
      avatar: undefined,
      rspId: 'TEST_RSP_ID',
      userId: user.id,
    });
    expect(dispatch).toBeCalled();
  });
});
