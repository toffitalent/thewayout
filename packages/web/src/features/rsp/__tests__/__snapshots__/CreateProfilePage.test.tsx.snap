// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProfilePage renders correctly 1`] = `
<div>
  <div
    class="wizardWrapper"
  >
    <section
      class="layout"
    >
      <div
        class="container"
      >
        <header
          class="header"
        >
          <button
            aria-label="Previous"
            class="button sm outline c-grey disabled iconButton sm round back backHidden"
            disabled=""
            type="button"
          >
            <svg
              aria-hidden="true"
              class="icon"
              focusable="false"
              stroke-width="2"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
                stroke="none"
              />
              <polyline
                points="15 6 9 12 15 18"
              />
            </svg>
          </button>
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="progress md rounded trackColor progress"
            role="progressbar"
          >
            <div
              class="track bgcolor-primary"
              style="transform: translateX(-100%);"
            />
          </div>
          <h2
            class="text title"
          >
            What's Your Phone Number?
          </h2>
        </header>
        <div
          class="maxWidth-96 marginRight-auto marginLeft-auto"
        >
          <form>
            <div
              class="control"
              role="group"
            >
              <label
                class="label"
                for=":r0:"
              >
                Phone number
              </label>
              <div
                class="root primary md fluid md"
              >
                <input
                  class="input"
                  id=":r0:"
                  maxlength="14"
                  name="phone"
                  placeholder="Add your phone number"
                  type="text"
                  validate="[object Object]"
                  value=""
                />
              </div>
            </div>
            <button
              class="button md solid c-primary disabled fluid maxWidth-96 marginTop-8"
              disabled=""
              type="submit"
            >
              Save & Next
            </button>
          </form>
        </div>
      </div>
    </section>
  </div>
</div>
`;
